opengovsg/FormSG
# refactor: duplicate adminform submissions to /api/v3 endpoint

## Problem
Migrates old endpoints located at `/:formId/adminform` to new endpoint located at `/api/v3/admin/forms` 

Closes #1511 

## Solution
1. shifts old endpoints rooted at `/adminform/submissions` to new api root 
2. combined them under a single `AdminFormsSubmissionsRouter`. 

**Improvements**:
1. Combined validator middleware together with their handlers. The reasoning for this is that certain handlers require parameters to be valid before the handler can proceed and hence, the handler should be exported atomically together with the middleware. 
2. For such handlers, comments have been added onto the base function to denote that they are exported **soley** for testing purposes 

## Tests

1. Ported over existing tests from `adminform.routes.spec.ts` for `/count` and `/download`

## Manual Tests 
1. Create a storage mode form with attachment fields. Publish the form, go to it and submit a few times with attachments.
    - Go to the admin panel and click on data. There should be `x responses to date`. Click on export and perform both actions. The files should be downloaded successfully. 
    - Change the date range to a date in the past without submissions. Verify that you are unable to export csv (both with/without attachments) 
    - Change the date change to a valid date range and try to export csv again (with and without). The files should be downloaded successfully
